Remove unused `TriggerRule.DUMMY` (#43349)
mistercrunch/Airflow: newsfragments/43368.significant.rst

Deprecated trigger rule removed

**Breaking Change**

> The trigger rule prior to TriggerRule.ALWAYS has been deprecated and now removed.

Reviewer: ```suggestion
The trigger rule prior to TriggerRule.DUMMY was removed.
```

Author: @eladkal I had to use the phrasing "prior to TriggerRule.ALWAYS" because the pre-commit prevents using the word `dummy`

Reviewer: Then we need to fix the pre commit. You can add exclusion for this specific file
